File: app/src/main/java/fobo66/exchangecourcesbelarus/util/CurrencyBuilderImpl.java
package fobo66.exchangecourcesbelarus.util;

import com.google.common.base.CaseFormat;

import java.lang.reflect.Field;

import fobo66.exchangecourcesbelarus.models.Currency;

/**
 * (c) 2017 Andrey Mukamolov aka fobo66 <fobo66@protonmail.com>
 * Created by fobo66 on 15.03.2017.
 */

public class CurrencyBuilderImpl implements CurrencyBuilder {
    private Currency currency;

    public CurrencyBuilderImpl() {
        this.currency = new Currency();
    }

    public CurrencyBuilderImpl with(String fieldName, String fieldValue) throws NoSuchFieldException, IllegalAccessException {
        Field field = Currency.class.getDeclaredField(convertFieldName(fieldName));
        field.setAccessible(true);
        field.set(currency, fieldValue);
        return this;
    }

    private String convertFieldName(String fieldName) {
        return CaseFormat.LOWER_UNDERSCORE.to(CaseFormat.LOWER_CAMEL, fieldName);
    }

    @Override
    public Currency build() {
        return currency;
    }
}

File: app/src/main/java/fobo66/exchangecourcesbelarus/util/CurrencyBuilderImpl.kt
package fobo66.exchangecourcesbelarus.util

import fobo66.exchangecourcesbelarus.entities.Currency

/**
 * (c) 2017 Andrey Mukamolov aka fobo66
 * Created by fobo66 on 15.03.2017.
 */
class CurrencyBuilderImpl : CurrencyBuilder {
  private val currencyMap = mutableMapOf<String, String>()

  override fun with(fieldName: String, fieldValue: String): CurrencyBuilder {
    currencyMap.put(fieldName, fieldValue)
    return this
  }

  override fun build(): Currency {
    return Currency(
      currencyMap["bankname"].orEmpty(),
      currencyMap["usd_buy"].orEmpty(),
      currencyMap["usd_sell"].orEmpty(),
      currencyMap["eur_buy"].orEmpty(),
      currencyMap["eur_sell"].orEmpty(),
      currencyMap["rur_buy"].orEmpty(),
      currencyMap["rur_sell"].orEmpty()
    )
  }
}